managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2366,6 +2402,9 @@ private void maybeOffload(CompletableFuture<PositionImpl> finalPromise) {
             unlockingPromise.whenComplete((res, ex) -> {
                     offloadMutex.unlock();
                     if (ex != null) {
+                        if (FutureUtil.unwrapCompletionException(ex) instanceof ManagedLedgerException) {
```

Comment:
Looks like we don't need check here? `asyncRefreshLedgersInfoOnBadVersion` already checked exception type

Reply:
Because the exception may a CompletionException, we need to make sure it is a ManagedLedgerException.